packages/docs/docs/fhir-datastore/patient-deduplication/patient-deduplication.md

```diff
@@ -0,0 +1,193 @@
+
+Typically, these pipelines compute matches in a data warehouse, compute engine, but can also can also be computed in a Medplum Bot with sufficient memory. A typical workflow is:
+
+1. Export patient data from Medplum into the appropriate data warehouse or compute engine (e.g. [Spark](https://spark.apache.org/)). Note that even large patient datasets should be able to fit into local memory (1M patients < 10GB), so distributed computation is not strictly required. See our [analytics guide](/docs/analytics) for more info.
+2. Use [matching rules](#matching-rules) to detect pairs of records. Because this is an N<sup>2</sup> operation, we recommend using some form of exact matching rules to reduce the cardinality of the problem, before applying "fuzzy matching."
+3. Use [merging rules](#merge-rules) to _create_ the master record.
+4. Use the Medplum API to update the `Patient.active` and `Patient.link` elements for all records.
+
+### Incremental Pipelines
+
+Each invocation of an incremental pipeline considers a single record and finds all matches. As these pipelines are typically used to manage high-frequency, event-driven updates, latency is more of a concern than memory.
+
+Incremental pipelines can often be implemented using [Medplum Bots](/docs/bots), and a typical workflow is:
+
+1. Set up a Bot to listen for updates to patient records.
+2. Use [matching rules](#matching-rules) to detect matching records. Because incremental pipelines only consider matches for a single record, we are less memory constrained and can apply more "fuzzy matching" rules.
+
+- Use [merging rules](#merge-rules) to _update_ the master patient record.
+
+- Update the `Patient.active` and `Patient.link` elements for all relevant records.
+
+  Check out [this blog post](/blog/patient-deduplication) for more details on event-driven pipelines. The medplum-demo-bots repo also contains an [example](https://github.com/medplum/medplum-demo-bots/blob/main/src/examples/patient-deduplication.ts) of an incremental deduplication bot.
+
+## Matching Rules {#matching-rules}
+
+The best deduplication systems use a library of matching rules with different strengths and weaknesses. While the effectiveness of different patient matching rules will vary depending on the clinical context, here we suggest some rules to get you started.
+
+These have been trialed in previous deduplication projects, and are rated by their false positive rates (i.e. incorrect matches) and false negative rates (i.e. missed matches).
+
+1. **Exact match on name, gender, email address / phone number: ** We recommend starting here. Email and phone act as pseudo-unique identifiers, and have a very low false positive rate. Using name and gender help eliminate false positives caused by family members sharing the same contact info. Note that phone numbers should be normalized before use.
```

Comment:
I'd combine 1 and 2 here.  At minimum, "date of birth" should be added to 1.  Very common false positive duplicate is John Jr taking care of John Sr in the same home, and handling all comms.